ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRao.java

```diff
@@ -55,70 +55,69 @@ public String getVersion() {
-        InitialSituation initialSituation = new InitialSituation(network, variantId, crac);
-        systematicAnalysisEngine.run(initialSituation);
+                                            LinearRaoParameters linearRaoParameters) {
+        String initialVariantId = situation.getWorkingVariantId();
+        systematicAnalysisEngine.run(situation);
 
         // stop here if no optimisation should be done
-        if (skipOptim(linearRaoParameters, crac)) {
-            return CompletableFuture.completedFuture(buildSuccessfulRaoResult(initialSituation, initialSituation, systematicAnalysisEngine));
+        if (skipOptim(linearRaoParameters, situation.getCrac())) {
+            return CompletableFuture.completedFuture(buildSuccessfulRaoResultAndClearVariants(situation, initialVariantId, initialVariantId, systematicAnalysisEngine));
         }
 
-        // start optimisation loop
-        AbstractSituation bestSituation = initialSituation;
+        String bestVariantId = initialVariantId;
+        String optimizedVariantId;
+
         for (int iteration = 1; iteration <= linearRaoParameters.getMaxIterations(); iteration++) {
+            optimizedVariantId = situation.cloneVariant(bestVariantId);
+            situation.setWorkingVariant(optimizedVariantId);
 
-            // look for a new RangeAction combination, optimized with the LinearOptimisationEngine
-            OptimizedSituation optimizedSituation = linearOptimisationEngine.run(bestSituation);
+            // Look for a new RangeAction combination, optimized with the LinearOptimisationEngine
+            // Stores found solutions in crac extension working variant
+            // Apply remedial actions on the network working variant
+            linearOptimisationEngine.run(situation);
 
             // if the solution has not changed, stop the search
-            if (bestSituation.sameRaResults(optimizedSituation)) {
-                optimizedSituation.deleteCracResultVariant();
-                optimizedSituation.deleteNetworkVariant();
+            if (situation.sameRemedialActions(bestVariantId, optimizedVariantId)) {
                 break;
             }
 
             // evaluate sensitivity coefficients and cost on the newly optimised situation
-            systematicAnalysisEngine.run(optimizedSituation);
-
-            if (optimizedSituation.getCost() < bestSituation.getCost()) { // if the solution has been improved, continue the search
-                if (!(bestSituation instanceof InitialSituation)) {
-                    bestSituation.deleteCracResultVariant();
-                    bestSituation.deleteNetworkVariant();
-                }
-                bestSituation = optimizedSituation;
+            systematicAnalysisEngine.run(situation);
+
+            double previousLowestCost = situation.setWorkingVariant(bestVariantId).getCost();
+            double optimizedCost = situation.setWorkingVariant(optimizedVariantId).getCost();
```

Comment:
I would have a bestCost variable and an optimizedCost variable that get updated like the bestVariantId and optimizedVariantId

Reply:
ok